format/postgres/common/pg_heap/postgres/pg_heap.go

```diff
@@ -0,0 +1,413 @@
+/*    8      |     2 */ // uint16 pd_checksum;
+/*   10      |     2 */ // uint16 pd_flags;
+/*   12      |     2 */ // LocationIndex pd_lower;
+/*   14      |     2 */ // LocationIndex pd_upper;
+/*   16      |     2 */ // LocationIndex pd_special;
+/*   18      |     2 */ // uint16 PdPagesizeVersion;
+/*   20      |     4 */ // TransactionId pd_prune_xid;
+/*   24      |     0 */ // ItemIdData pd_linp[];
+//
+/* total size (bytes):   24 */
+
+// type = struct PageXLogRecPtr {
+/*    0      |     4 */ // uint32 xlogid;
+/*    4      |     4 */ // uint32 xrecoff;
+
+/* total size (bytes):    8 */
+
+// typedef uint16 LocationIndex;
+// #define SizeOfPageHeaderData (offsetof(PageHeaderData, pd_linp))
+
+// type = struct HeapTupleHeaderData {
+/*    0      |    12 */ // union {
+/*                12 */ //     HeapTupleFields t_heap;
+/*                12 */ //     DatumTupleFields t_datum;
+//
+/* total size (bytes):   12  */
+/* } t_choice;             //
+/*   12      |     6 */ // ItemPointerData t_ctid;
+/*   18      |     2 */ // uint16 t_infomask2;
+/*   20      |     2 */ // uint16 t_infomask;
+/*   22      |     1 */ // uint8 t_hoff;
+/*   23      |     0 */ // bits8 t_bits[];
+/* XXX  1-byte padding  */
+//
+/* total size (bytes):   24 */
+const SizeOfHeapTupleHeaderData = 24
+
+// type = struct HeapTupleFields {
+/*    0      |     4 */ // TransactionId t_xmin;
+/*    4      |     4 */ // TransactionId t_xmax;
+/*    8      |     4 */ // union {
+/*                 4 */ //    CommandId t_cid;
+/*                 4 */ //    TransactionId t_xvac;
+//                         } t_field3;
+/*                      total size (bytes):    4 */
+//
+/* total size (bytes):   12 */
+
+// type = struct DatumTupleFields {
+/*    0      |     4 */ // int32 datum_len_;
+/*    4      |     4 */ // int32 datum_typmod;
+/*    8      |     4 */ // Oid datum_typeid;
+//
+/* total size (bytes):   12 */
+
+// type = struct ItemPointerData {
+/*    0      |     4 */ // BlockIdData ip_blkid;
+/*    4      |     2 */ // OffsetNumber ip_posid;
+//
+/* total size (bytes):    6 */
+
+type Heap struct {
+	Args format.PostgresHeapIn
+
+	// current Page
+	Page *HeapPage
+	// Page special data
+	Special *PageSpecial
+
+	// current tuple
+	Tuple *TupleD
+
+	DecodePageHeaderData func(page *HeapPage, d *decode.D)
+	DecodePageSpecial    func(heap *Heap, d *decode.D)
+}
+
+type PageSpecial struct {
+	// pgproee
+	PdXidBase   uint64 // 8 TransactionId pd_xid_base;
+	PdMultiBase uint64 // 8 TransactionId pd_multi_base;
+	PdPruneXid  uint64 // 4 ShortTransactionId pd_prune_xid;
+	PdMagic     uint64 // 4 uint32 pd_magic;
+}
+
+type TupleD struct {
+	IsMulti bool
+}
+
+func Decode(heap *Heap, d *decode.D) any {
+	decodeHeapPages(heap, d)
+	return nil
+}
+
+func decodeHeapPages(heap *Heap, d *decode.D) {
+	blockNumber := uint32(heap.Args.Page + heap.Args.Segment*common.RelSegSize)
+	count := int64(0)
+	for {
+		if end, _ := d.TryEnd(); end {
```

Comment:
Same as the other comment about `SeekAbs/TryEnd`, can be just `End()`?

Reply:
It's done.